lib/dtutils.lua

```diff
@@ -0,0 +1,457 @@
+--[[
+
+    dtutils.lua - common darktable lua functions
+
+    Copyright (C) 2016 Bill Ferguson <wpferguson@gmail.com>.
+    Copyright (C) 2016 Tobias Jakobs
+
+    This program is free software: you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation; either version 3 of the License, or
+    (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+]]
+
+local dtutils = {}
+
+dtutils.libdoc = {
+  Sections = {"Name", "Synopsis", "Description", "See_Also", "License"},
```

Comment:
I modeled the documentation after man pages.  At the same time I wanted to have the flexibility to add or remove sections, so it's not a "strict" man page format.  The only way I could figure out to get the sections to come out in order was using the array.  If I just used keys then the order isn't guaranteed.  It can be error prone.  I got caught once with a section and no key, which is why there's a check in the tools to make sure section doesn't reference a non-existent key.

Adding the copyright section is simply a matter of adding it to the array, then adding the key and value.  I'll do that.